submissions/unabyband/oop_exercise/index.js

```diff
@@ -0,0 +1,102 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/unabyband/a-tiny-JS-world
+   Web app: https://unabyband.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+  constructor(species, name, gender, legs, hands, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.legs = legs;
+    this.hands = hands;
+    this.saying = saying;
+    this.friends = friends;
+  }
+
+  printInhabitant() {
+    const nameToPrint = `<strong>${this.name}</strong>`;
+    const talkToPrint = `<em>${this.saying}</em>`;
+    const friendsToPrint = this.friends
+      ? this.friends.map((friend) => friend.name).join(", ")
+      : "Cats have no friends";
+    const limbs =
+      this.hands != 0
+        ? `On ${this.legs} legs and has ${this.hands} hands`
+        : `On ${this.legs} paws`;
+
+    print(
+      [
+        this.species,
+        nameToPrint,
+        this.gender,
+        limbs,
+        talkToPrint,
+        friendsToPrint,
+      ].join("; ")
+    );
+  }
+}
+
+class Human extends Inhabitant {
+  constructor(name, gender, saying, friends) {
+    super("human", name, gender, 2, 2, saying, friends);
+  }
+}
+
+class Animal extends Inhabitant {
```

Comment:
In fact, every instance of `Animal` has property for hands - `hands`.
OOP helps to design the system of business entities so that they just do not have unnecessary properties.
This is something bigger than just a way to provide default values to the base class.

Also requirement say that inhabitants that do not have something by natural design they do not have any properties for this. OOP helps to build scalable code that doesn't require to change base class when we want to handle something specific to a sub-class.

Make so that only humans have hands in any meaning. Also remember that very class should handle the properties it owns immediately. Do not delegate property handling to parent class (it should be agnostic of any properties a child class may have).

Reply:
Well, I've remade legs/hands/paws properties for ancestor and child classes. Now the Animal class doesn't have any type of 'hands' property.
For best (in my opinion) improvement - I decided to pass into legs property of ancestor class already finished string contains formatted out.
Though, I also thought about rewriting printInhabitant method especially for Human class, just in order to printing hands and legs unlike the named method in ancestor class, but considered this way less DRY (Even though rewriting methods is one of the OOP-principles). 
Please re-review my fixes. Thank you.
And please, excuse me for long feedback time, I had to finished current work.